pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -235,10 +235,12 @@ func configureFromProviderConfig(pushCtx *model.PushContext, meta *model.NodeMet
 type typedConfigGenFromClusterFn func(clusterName string) (*anypb.Any, error)
 
 func zipkinConfigGen(cluster string) (*anypb.Any, error) {
+	_, _, hostname, _ := model.ParseSubsetKey(cluster)
 	zc := &tracingcfg.ZipkinConfig{
 		CollectorCluster:         cluster,
 		CollectorEndpoint:        "/api/v2/spans",                   // envoy deprecated v1 support
 		CollectorEndpointVersion: tracingcfg.ZipkinConfig_HTTP_JSON, // use v2 JSON for now
+		CollectorHostname:        string(hostname),                  // http host header
```

Comment:
The hostname from the cluster is not appropriate for setting the upstream Hostname in all cases; it may he a wildcard, etc. While this is reasonable for k8s Services I am worried about the behavior on ServiceEntry

Reply:
SE with wildcard is very rare, especially for case here.

Whatever, it is better than before.